src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -1459,13 +1544,14 @@ private class FactoryContext
         public bool AllowEmptyRequestBody { get; set; }
 
         public bool UsingTempSourceString { get; set; }
-        public List<ParameterExpression> ExtraLocals { get; } = new();
-        public List<Expression> ParamCheckExpressions { get; } = new();
+        public List<ParameterExpression> Locals { get; } = new();
+        public List<Expression> Expressions { get; } = new();
```

Comment:
Why did you rename this? All the expressions in this list are still doing some sort of parameter validation, right?

Reply:
Because the names are bad. These aren't really about checking parameters its just a big old block being built with locals and statements.

Reply:
I'd buy that if and when we use this for general expressions, but the new naming implies that these are all the expressions being compiled even though most of the expressions (those returned by `CreateArgument()`) don't end up in this list.

`ParamCheckExpressions` is only used by `CreateParamCheckingResponseWritingMethodCall` to include additional expressions that all include `Expression.Assign(WasParamCheckFailureExpr, Expression.Constant(true))`. We check `factoryContext.ParamCheckExpressions.Count > 0` to determine if we need to add the following to the returned `RequestDelegate`:

```C#
Expression.Condition(WasParamCheckFailureExpr,
            set400StatusAndReturnCompletedTask,
            AddResponseWritingToMethodCall(methodCall, methodInfo.ReturnType))
```

This makes absolutely no sense unless all the expressions that could end up in the list are doing parameter checking. I agree this mechanism could be expanded and used more generally with further modification (mainly using some other state to determine if we're doing param checking), but I don't see the point of removing the self-documentation until then.

Reply:
Got it, I reverted the name change.